go/client/tree.go

```diff
@@ -21,6 +25,77 @@ type FileTree struct {
 	Dirs  map[string]*FileTree
 }
 
+func shouldIgnore(is *command.InputSpec, inp string, t command.InputType) bool {
+	for _, r := range is.InputExclusions {
+		if r.Type != command.UnspecifiedInputType && r.Type != t {
+			continue
+		}
+		if m, _ := regexp.MatchString(r.Regex, inp); m {
+			return true
+		}
+	}
+	return false
+}
+
+func collectFiles(execRoot string, is *command.InputSpec, path string, fs map[string][]byte) error {
```

Comment:
You only need exclusions here not the full inputspec, correct?

Reply:
Right, changed.